backend/contracts/Quiz.sol

```diff
@@ -334,51 +390,58 @@ contract Quiz {
             revert InvalidCouponError(pc.coupon);
         }
 
-        // Make a payout.
-        (bool success, ) = pc.addr.call{value: _reward}("");
-        if (!success) {
-            revert PayoutFailedError();
+        // If nft address is set and , mint NFT
+        if (_nftAddress != address(0)) {
+            OasisReward(getNftAddress()).mint(pc.addr, _jsonHash);
         }
-
-        if (_makePayoutNFT) {
-            // Mint NFT
-            OasisReward(getNft()).mint(pc.addr, _svgImage);
+        // If _reward is set, make a payout.
+        if (this.isReward()) {
+            (bool success, ) = pc.addr.call{value: _reward}("");
+            if (!success) {
+                revert PayoutFailedError();
+            }
         }
 
         // Invalidate coupon.
         _coupons[pc.coupon] = block.number;
-    
+
         // Increase nonce, for gasless tx.
         if (msg.sender == _kp.addr) {
             _kp.nonce++;
         }
     }
 
-    // Sets the flag for making a ROSE payout and the flag for making a NFT payout
-    function setPayoutFlags(
-        bool makePayoutNative,
-        bool makePayoutNFT
-    ) external onlyOwner {
-        _makePayoutNative = makePayoutNative;
-        _makePayoutNFT = makePayoutNFT;
-    }
-
-    // Adds the IDs of the NFT contract to the Quiz contract
-    function setNft(address nftAddress) external onlyOwner {
-        _quizID = uint32(block.number);
-        _svgImage = OasisReward(nftAddress).generateComplexSVG(_quizID);
+    /**
+     * @notice Sets the NFT contract address and
+     *  generates the SVG image for the quiz.
+     * @param nftAddress NFT contract address.
+     */
+    function setNftAddress(address nftAddress) external onlyOwner {
         _nftAddress = nftAddress;
     }
 
-    // Function to get the deployed ERC721 contract instance
-    function getNft() public view returns (address) {
+    /**
+     * @notice Returns the NFT contract address.
+     */
+    function getNftAddress() public view returns (address) {
         if (_nftAddress == address(0)) {
             revert ZeroAddressError(_nftAddress);
         }
         return _nftAddress;
     }
 
-    // Reclaims contract funds to given address.
+    /**
+     * @notice Sets the custom json metadata for the quiz.
+     * @param jsonMetadata JSON metadata to add to the OasisReward contract.
+     */
+    function setCustomNFT(string calldata jsonMetadata) external onlyOwner {
```

Comment:
I think this can simply become storeNFT? Also fix in the hardhat.config CLI.

```suggestion
    function storeNFT(string calldata jsonMetadata) external onlyOwner {
```

Reply:
Renamed to storeNFT, updated the hardhat.config